hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ReconTaskControllerImpl.java

```diff
@@ -263,7 +266,7 @@ private void processTasks(
               taskStatusUpdaterManager.getTaskStatusUpdater(taskName);
           if (!result.getRight().getRight()) {
             LOG.error("Task {} failed", taskName);
-            failedTasks.add(new ImmutablePair<>(result.getLeft(), result.getRight().getLeft()));
+            failedTasks.add(new ImmutablePair<>(taskName, result.getRight().getLeft()));
```

Comment:
IMO, the result TaskResult as pojo can be create, instead of going multiple level pair and other data structure in it. Pojo can be pre-init with Collections.emptyMap() to optimize memory.
This will help in redability and changes to all methods.